src/solver.jl

```diff
@@ -28,16 +28,16 @@
 end
 
 function POMDPs.solve(solver::DeepQLearningSolver, problem::MDP)
-    env = MDPEnvironment(problem, rng=solver.rng)
+    env = MDPCommonRLEnv{AbstractArray{Float32}}(problem) # ignores solver.rng because CommonRLEnv doesn't have rng support yet
     return solve(solver, env)
 end
 
 function POMDPs.solve(solver::DeepQLearningSolver, problem::POMDP)
-    env = POMDPEnvironment(problem, rng=solver.rng)
+    env = POMDPCommonRLEnv{AbstractArray{Float32}}(problem) # ignores solver.rng because CommonRLEnv doesn't have rng support yet

```

Comment:
We can't get any more concrete information except `AbstractArray{Float32}`?

Reply:
I think `AbstractArray{Float32}` is what we want here. It just means that `convert_o(AbstractArray{Float32}, o, pomdp)` will be called on every observation. That way problem-writers can use static arrays or built-in arrays. If the problem implementation is type-stable, the compiler should still be able to infer a nice concrete return type.